charts/kube-prometheus-stack/templates/prometheus-operator/servicemonitor.yaml

```diff
@@ -9,7 +9,19 @@ metadata:
 {{ include "kube-prometheus-stack.labels" . | indent 4 }}
 spec:
   endpoints:
+  {{- if .Values.prometheusOperator.tls.enabled }}
+  - port: https
+    scheme: https
+    tlsConfig:
+      serverName: {{ template "kube-prometheus-stack.operator.fullname" . }}
+      ca:
+        secret:
+          name: {{ template "kube-prometheus-stack.fullname" . }}-admission
```

Comment:
Where is this secret defined? It does not seem to exist in GKE with admission webhooks disabled.

> caller=operator.go:1628 component=prometheusoperator msg="skipping servicemonitor" error="failed to get CA: unable to get secret \"prometheus-operator-kube-p-admission\": secrets \"prometheus-operator-kube-p-admission\" not found" servicemonitor=monitoring/prometheus-operator-kube-p-operator namespace=monitoring prometheus=prometheus-operator-kube-p-prometheus

Reply:
It's create by this job: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/job-createSecret.yaml when admission webhooks is enable. This was the same secret used by the TLS proxy.

Reply:
Maybe this should check for `if .Values.prometheusOperator.admissionWebhooks.enabled` to avoid this "unable to get secret error"? :thinking: